runtime/syntax/c.yaml

```diff
@@ -23,10 +23,11 @@ rules:
     - symbol.operator: "[-+*/%=<>.:;,~&|^!?]|\\b(offsetof|sizeof)\\b"
     - symbol.brackets: "[(){}]|\\[|\\]"
       # Integer Constants
-    - constant.number: "(\\b([1-9][0-9]*|0[0-7]*|0[Xx][0-9A-Fa-f]+|0[Bb][01]+)([Uu][Ll]?[Ll]?|[Ll][Ll]?[Uu]?)?\\b)"
-      # Decimal Floating Constants
-    - constant.number: "(\\b(([0-9]*[.][0-9]+|[0-9]+[.][0-9]*)([Ee][+-]?[0-9]+)?|[0-9]+[Ee][+-]?[0-9]+)[FfLl]?\\b)"
+    - constant.number: "(\\b([1-9][0-9']*|0[0-7']*|0[Xx][0-9a-fA-F']+|0[Bb][01']+)([Uu]?[Ll][Ll]?|[Ll][Ll]?[Uu]?)?\\b)"
```

Comment:
Regarding the switch of the order in the list:
There are still places in the file where it's written the other way around.

Reply:
I have reverted the accidental changes for the order

Reply:
@dmaluka 
Yeah I was just copying from cpp.yaml cuz I thought these lines are the same (they looked similar) so I just copied and pasted.

I can fix whichever is wrong in this or another PR if needed

Reply:
Sure, I will apply the ones in `c.yaml` to `cpp.yaml` and also see if I can get 1 and 2 working

Reply:
First of all, I am not sure what to do with floating points since the part before or after the decimal are optional (i.e. `1.f` or `.1f` and hex/bin equivalent). Enforcing the rules are going to make the already long regex be longer I think. I am inclined to just leave it as it is.

As for non floating points, I can enforce 1 of them but not both in one regex. If I do both, it will look like this.
`0[Bb][01][01']*[01].

I will have to span each representation into multiple lilnes if I have to enforce these 2 rules. Unless you have another way of doing it. I am not the best at regex tbh :sweat_smile:

Reply:
> I will have to span each representation into multiple lilnes if I have to enforce these 2 rules. Unless you have another way of doing it. I am not the best at regex tbh 😅

As you can see, you can specify multiple regexes for `constant.number` instead of one very long regex. There are already 3 of them, you can split them further.

Reply:
I have broken them down for each scenario. It should work now.